File: programmers/코딩테스트 고득점 Kit/완전탐색/카펫.cpp
#include <string>
#include <vector>
using namespace std;
vector<pair<int, int>> candicate;
vector<int> solution(int brown, int yellow) {
	vector<int> ans;
	for (int i = 1; i <= yellow; i++) {
		if (yellow % i == 0) {
			if (i < (yellow / i)) continue;
			candicate.push_back({ i, yellow / i });
		}
	}
	for (int i = 0; i < candicate.size(); i++) {
		if ((candicate[i].first + 2) * (candicate[i].second + 2) == (yellow + brown)) {
			ans.push_back(candicate[i].first + 2);
			ans.push_back(candicate[i].second + 2);
			return ans;
		}
	}
}

File: programmers/코딩테스트 고득점 Kit/완전탐색/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int total = brown + yellow;
        for(int i=1; i <= (int) Math.sqrt(total); i++) {
            if(total % i != 0) continue;
            int width = total / i;
            int height = i;
            if(width * 2 + (height - 2) * 2 == brown) return new int[]{width, height};
        }
        return null;
    }
}